ui/app/store/actions.js

```diff
@@ -583,6 +692,66 @@ export function signTx (txData) {
   }
 }
 
+export function approveAllowance (allowance) {
+  return async (dispatch, getState) => {
+    const state = getState()
+
+    const { address: tokenAddress, decimals } = getSwapFromAsset(state)
+    const allowanceTarget = getSwapQuote(state).allowanceTarget
+
+    const maxAllowance = ethers.BigNumber.from(2).pow(256).sub(1)
+    const decimalFactor = ethers.BigNumber.from(10).pow(decimals)
+    const computedAllowance = allowance ? ethers.BigNumber.from(allowance).mul(decimalFactor) : maxAllowance
```

Comment:
is it expected for allowance to be empty here?

Reply:
just wondering if it should throw an error instead of being set to maxAllowance in that case

Reply:
i saw the note in https://github.com/brave/ethereum-remote-client/pull/242 about the max allowance. my preference would be to error if no value is set. however i defer to you on whether this would be too much of a usability hurdle since i'm not sure what users expect.

Reply:
After exploring different dapps, I think we should either set the allowance to the value of the swap or the available balance. The max allowance must be explicitly chosen if the user so wishes.

I'm going to push a fix for the default allowance.

Reply:
> is it expected for allowance to be empty here?
> just wondering if it should throw an error instead of being set to maxAllowance in that case

Sorry, I forgot to respond to the above - an empty allowance means the user chose the value proposed as default. The `allowance` value is not empty only if the user provided a custom value.

Also, I've pushed the following fixes:
- use the swap `amount` as the default proposed limit, with a fallback to the token balance.
- added a warning tooltip, nudging the user to set approval allowance to 0 first. See discussion [here](https://github.com/brave/security/issues/480#issuecomment-865581044).